TestCases/compliance-level-3/0068-feel-equality/0068-feel-equality.dmn

```diff
@@ -803,6 +803,16 @@
         </literalExpression>
     </decision>
 
+    <decision name="dt_duration_008" id="_dt_duration_008">
+        <description>Tests FEEL expression: 'duration("P0D") = duration("P0Y")' and expects result: 'true (boolean)'</description>
+        <question>Result of FEEL expression 'duration("P0D") = duration("P0Y")'?</question>
+        <allowedAnswers>true (boolean)</allowedAnswers>
+        <variable typeRef="boolean" name="dt_duration_008"/>
+        <literalExpression>
+            <text>duration("P0D") = duration("P0Y")</text>
```

Comment:
Unless I am mistaken, the paragraph mentioned in the original Description while PR raised, pertains to
```
9.4 S-FEEL semantics
```

and does not correspond to FEEL semantics.

Friendly reminder, S-FEEL has been deprecated at RTF.

My understanding of FEEL semantic applicable here is:

Duration built-in function:
```
duration(from) | duration string | convert from to a days and time or years and months duration 
```

meaning:
```
duration("P0D") // an instance of days and time duration
```
then
```
duration("P0Y") // an instance of years and months duration
```

and accordingly to type lattice:
![image](https://user-images.githubusercontent.com/1699252/106583974-f2778500-6545-11eb-90cd-0833b8890e30.png)

they are not the same type.

For as much as I believe this is one of the non-business-friendly area of FEEL, my understanding is that the result of this test is `null`, since not complying with the definition for:

```
Table 52: General semantics of equality and inequality
...
49.a | e1 = e2 |   e1 and e2 must both be of the same kind/datatype – both numbers, both strings, etc.
...

Reply:
Thanks @tarilabs. Yes, i saw it was in the sfeel section, but im not sure that means it should be discarded. Some other things in that section are respected outside of sfeel. The definitions of arithmetic for example, like  op:add-dayTimeDuration etc. They are only discussed in sfeel section but apply in general. 

I was kind of expecting some discussion about this so comments welcome. Im not religious either way but my reading of it was it was not just sfeel, even though it is in that section. And the specific mention of an exception to the type lattice is worth a discussion. 👍

Reply:
Agree with @tarilabs on this

Reply:
just playing "devil's advocate" here. Should the "op:add-dayTimeDuration" (etc) explanations in that section also not apply just because they in the sfeel section?  My personal feeling is that the equality thing does apply, but maybe it is best we seek clarification as to whether it is intended to be sfeel only.  The spec is not exactly a shining example of unambiguous clarity.  To my mind, it would seem odd to make sfeel have a superset of some feel functionality.

Reply:
> The semantics of S-FEEL expressions are defined in this section, in terms of the semantics of the XML Schema datatypes and the XQuery 1.0 and XPath 2.0 Data Model datatypes, and in terms of the corresponding functions and operators defined by XQuery 1.0 and XPath 2.0 Functions and Operators (prefixed by “op:” below). **_A complete stand-alone specification of the semantics is to be found in clause 10.3.2, as part of the definition of FEEL. Within the scope of SFEEL,the two definitions are equivalent and equally normative._**

I have to say, I'm not sure exactly what this is saying, but I _think_ it says the operations described for SFEEL and FEEL are the same.  It describes duration equality as conforming to `op:duration-equal` and that has a specific example here https://www.w3.org/TR/xpath-functions-31/#func-duration-equal that shows it:

> The expression op:duration-equal(xs:yearMonthDuration("P0Y"), xs:dayTimeDuration("P0D")) returns true().

Just sayin' ....